src/asm/section.c

```diff
@@ -477,6 +477,21 @@ uint32_t sect_GetOutputOffset(void)
 	return curOffset + loadOffset;
 }
 
+// Returns how many bytes need outputting for the specified alignment and offset to succeed
+uint32_t sect_GetAlignBytes(uint8_t alignment, uint16_t offset)
+{
+	struct Section *sect = sect_GetSymbolSection();
+
+	// If the section is undefined, or not fixed and not aligned, no bytes are needed
+	if (!sect || (sect->org == (uint32_t)-1 && sect->align == 0))
+		return 0;
+
+	uint32_t pcValue = sect->org != (uint32_t)-1 ? sym_GetPCValue() : sect->align != 0
```

Comment:
Wouldn't you want to apply the modulo always? Otherwise the subtraction may yield something negative / underflow...

It should be fine to apply it even to the constant zero if it makes more semantic sense: the optimiser is likely smart enough to fold it if it so desires.

Reply:
No, that causes this test case to fail:

```asm
SECTION "fixed", ROM0[$100]
ds align[2] ; already aligned, 0 bytes
db 8, 9, 10
ds align[4, $e], 11 ; (1 << 4) - three - two = 11 bytes
db 12, 13 ; two
align 4
```

```
+error: ds-align.asm(10):
+    Section's fixed address fails required alignment (PC = $0111)
+error: ds-align.asm(12):
+    Section's fixed address fails required alignment (PC = $0113)
+error: Assembly aborted (2 errors)!
```